### PR TITLE
Duplo 11675 RDS Group Cluster Parameter support

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview
Support for Group Cluster Parameter for RDS resource

## Summary of changes
Added support to add cluster parameter group name for rds instance for aurora db's
This PR does the following:

-  Added `cluster_parameter_group_name` attribute to `duplo_rds_instance` schema
-  Setting `cluster_parameter_group_name` attribute in rdsInstanceToState()
-  Setting `cluster_parameter_group_name` attribute in rdsInstanceFromState()
-  Added example for `cluster_parameter_group_name`
- Updated doc

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
enhancement, documentation

___

## **Description**
- Implemented `duplocloud_gcp_k8_node_pools` resource with full schema and CRUD operations.
- Added `cluster_parameter_group_name` attribute to the RDS instance resource schema.
- Updated documentation and examples for RDS instance to include `cluster_parameter_group_name`.
- Added SDK support for GCP K8 Node Pools.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>provider.go</strong><dd><code>Register GCP Node Pools Resource in Provider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/provider.go

- Added `duplocloud_gcp_node_pools` to the provider resources.

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-0959744faba70cfc67abd70733e39a6f668d41c9fae69a27968841ef8ed56eeb">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_gcp_k8_node_pool.go</strong><dd><code>Implement GCP K8 Node Pools Resource</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_gcp_k8_node_pool.go

<li>Implemented the <code>duplocloud_gcp_k8_node_pools</code> resource with full schema <br>and CRUD operations.<br> <li> Added support for various configurations including machine type, disk <br>size, image type, tags, and autoscaling.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-bc0cc4a27038eb399598c526fd7148bd7cd9349310b06235a990a5f2cac1eaf9">+735/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_rds_instance.go</strong><dd><code>Support Cluster Parameter Group Name for RDS Instances</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_rds_instance.go

<li>Added <code>cluster_parameter_group_name</code> attribute to the RDS instance <br>resource schema.<br> <li> Included <code>cluster_parameter_group_name</code> in state functions for RDS <br>instances.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-a8bf9b56a3e9b39899a5d37d7608110678908754e0848fab2d0320653ea65f71">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_node_pools.go</strong><dd><code>Add SDK Support for GCP K8 Node Pools</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/gcp_node_pools.go

<li>Added SDK support for GCP K8 Node Pools including create, read, <br>update, and delete operations.<br> <li> Included structs for node pool configurations and accelerator <br>settings.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-d7d6988aa1383fd5e2f3442050fe77b783a7d1a7094e70e20982f64d9f1c5b03">+129/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rds_instance.go</strong><dd><code>Add Cluster Parameter Group Name to RDS Instance Struct</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/rds_instance.go

<li>Added <code>ClusterParameterGroupName</code> field to the <code>DuploRdsInstance</code> struct.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-3aeb597ae11ae40342366a786a3b78f55a1265ea03d8d2729401d9a7f3b7291a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update Changelog with RDS and GCP Node Pool Enhancements</code>&nbsp; </dd></summary>
<hr>
      
CHANGELOG.md

<li>Documented the addition of <code>cluster_parameter_group_name</code> for RDS <br>instances and the implementation of <code>duplocloud_gcp_node_pool</code> resource.<br> <br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rds_instance.md</strong><dd><code>Document Cluster Parameter Group Name for RDS Instances</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
docs/resources/rds_instance.md

<li>Updated RDS instance resource documentation to include <br><code>cluster_parameter_group_name</code>.<br> <li> Provided example usage for creating an RDS instance with <br><code>cluster_parameter_group_name</code>.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-2be785952795586c48f30db78e5f3684f72a430f66155dd8ea0d85f0607eedb7">+18/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource.tf</strong><dd><code>Add Example for RDS Instance with Cluster Parameter Group Name</code></dd></summary>
<hr>
      
examples/resources/duplocloud_rds_instance/resource.tf

<li>Added an example Terraform configuration for creating an RDS instance <br>with <code>cluster_parameter_group_name</code>.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/424/files#diff-4461dc184c959908eb1db51855cf42bfafd849c340d3632e8da923f5a5a6240d">+17/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11675 TF: Support RDS Cluster Parameter Group Name parameter](https://app.clickup.com/t/8655600/DUPLO-11675)

**Review by joek-duplo — CHANGES_REQUESTED** [review]
This PR should not have any code for node pools at all.  Please remove that and fix the changelog.

File: duplocloud/resource_duplo_rds_instance.go [hunk 1]
```diff
@@ -624,6 +636,9 @@ func rdsInstanceFromState(d *schema.ResourceData) (*duplosdk.DuploRdsInstance, e
 	duploObject.EngineVersion = d.Get("engine_version").(string)
 	duploObject.SnapshotID = d.Get("snapshot_id").(string)
 	duploObject.DBParameterGroupName = d.Get("parameter_group_name").(string)
+	if isAuroraDB(d) {
```
[the remark this answers is not in the document]
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_rds_instance.go:639]: isAuroraDB checks for multiple DB
DUPLO_RDS_ENGINE_AURORA_MYSQL                 = 8
DUPLO_RDS_ENGINE_AURORA_POSTGRESQL            = 9
DUPLO_RDS_ENGINE_AURORA_SERVERLESS_MYSQL      = 11
DUPLO_RDS_ENGINE_AURORA_SERVERLESS_POSTGRESQL = 12
AURORA = 16

other than aurora supported db's are
MySQL 
PostgreSQL
MsftSQL-Express
MsftSQL-Standard
MsftSQL-Web
DocumentDB
MariaDB

So should I remove the isAuroraDB check?

[the remark this answers is not in the document]
Comment (joek-duplo) [reply duplocloud/resource_duplo_rds_instance.go:639]: no, my complaint was not to remove the aurora check, it was to include any additional DB that also supports a cluster parameter group name.  please check with @pravin-duplocloud , I believe he knows the logic here.

**Comment by Duplo-nikv** [comment]
Yes I got the additional Db's talked with Praveen on this.

**Review by tahir-duplo — APPROVED** [review]
LGTM

File: duplocloud/resource_duplo_rds_instance.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_rds_instance.go]: Checked the UI for all databases which is associated with cluster group parameter along with @pravin-duplocloud 
https://github.com/duplocloud/terraform-provider-duplocloud/pull/424/commits/6f117b155ddd26ba6661b45b202c79cb7387a921

Nodepool code trace where removed

**Pull request merged**